code/modules/mob/living/simple_animal/friendly/drone/_drone.dm

```diff
@@ -181,9 +181,15 @@
 	if(default_storage)
 		var/obj/item/I = new default_storage(src)
 		equip_to_slot_or_del(I, ITEM_SLOT_DEX_STORAGE)
-	if(default_hatmask)
-		var/obj/item/I = new default_hatmask(src)
-		equip_to_slot_or_del(I, ITEM_SLOT_HEAD)
+
+	for(var/holiday_name in GLOB.holidays)
+		var/datum/holiday/holiday_today = GLOB.holidays[holiday_name]
+		if(holiday_today.holiday_hat && !default_headwear) //If our drone type doesn't start with a hat, we take the holiday one.
+			default_headwear = holiday_today.holiday_hat
```

Comment:
```suggestion
		var/datum/holiday/holiday_today = GLOB.holidays[holiday_name]
		var/obj/item/potential_hat = holiday_today.holiday_hat
		if(!isnull(potential_hat) && isnull(default_headwear)) //If our drone type doesn't start with a hat, we take the holiday one.
			default_headwear = potential_hat
```

`isnull()` for better explicitness, and cache the variable since we could potentially access it twice (just an easy clarity thing)